core/src/main/java/org/elasticsearch/common/Booleans.java

```diff
@@ -56,77 +50,148 @@ public static boolean isBoolean(char[] text, int offset, int length) {
         if (text == null || length == 0) {
             return false;
         }
-        if (length == 1) {
-            return text[offset] == '0' || text[offset] == '1';
-        }
-        if (length == 2) {
-            return (text[offset] == 'n' && text[offset + 1] == 'o') || (text[offset] == 'o' && text[offset + 1] == 'n');
-        }
-        if (length == 3) {
-            return (text[offset] == 'o' && text[offset + 1] == 'f' && text[offset + 2] == 'f') ||
-                    (text[offset] == 'y' && text[offset + 1] == 'e' && text[offset + 2] == 's');
-        }
-        if (length == 4) {
-            return (text[offset] == 't' && text[offset + 1] == 'r' && text[offset + 2] == 'u' && text[offset + 3] == 'e');
-        }
-        if (length == 5) {
-            return (text[offset] == 'f' && text[offset + 1] == 'a' && text[offset + 2] == 'l' && text[offset + 3] == 's' && text[offset + 4] == 'e');
-        }
-        return false;
+        return isBoolean(new String(text, offset, length));
+    }
+
+    public static boolean isBoolean(String value) {
+        return isFalse(value) || isTrue(value);
     }
 
-    /***
+    /**
+     * Parses a string representation of a boolean value to <code>boolean</code>.
      *
-     * @return true/false
-     * throws exception if string cannot be parsed to boolean
+     * @return <code>true</code> iff the provided value is "true". <code>false</code> iff the provided value is "false".
+     * @throws IllegalArgumentException if the string cannot be parsed to boolean.
      */
-    public static Boolean parseBooleanExact(String value) {
-        boolean isFalse = isExplicitFalse(value);
-        if (isFalse) {
+    public static boolean parseBoolean(String value) {
+        if (isFalse(value)) {
             return false;
         }
-        boolean isTrue = isExplicitTrue(value);
-        if (isTrue) {
+        if (isTrue(value)) {
             return true;
         }
+        throw new IllegalArgumentException("Failed to parse value [" + value + "] as only [true] or [false] are allowed.");
+    }
 
-        throw new IllegalArgumentException("Failed to parse value [" + value + "] cannot be parsed to boolean [ true/1/on/yes OR false/0/off/no ]");
+    /**
+     *
+     * @param value text to parse.
+     * @param defaultValue The default value to return if the provided value is <code>null</code>.
+     * @return see {@link #parseBoolean(String)}
+     */
+    public static boolean parseBoolean(String value, boolean defaultValue) {
+        if (value == null || value.length() == 0) {
+            return defaultValue;
+        }
+        return parseBoolean(value);
     }
 
     public static Boolean parseBoolean(String value, Boolean defaultValue) {
```

Comment:
Yes, there are some places indeed. For example, we use it when parsing REST request parameters in `RestActions#urlParamsToQueryBuilder(RestRequest)`.